src/wp-includes/abilities-api/class-wp-abilities-registry.php

```diff
@@ -78,11 +78,11 @@ final class WP_Abilities_Registry {
 	 * @return WP_Ability|null The registered ability instance on success, null on failure.
 	 */
 	public function register( string $name, array $args ): ?WP_Ability {
-		if ( ! preg_match( '/^[a-z0-9-]+\/[a-z0-9-]+$/', $name ) ) {
+		if ( ! preg_match( '/^[a-z0-9-]+\/([a-z0-9-]+\/)?[a-z0-9-]+$/', $name ) ) {
```

Comment:
According to the example, where the second path segment is `v1`, the regex would seem to rather be:

```suggestion
		if ( ! preg_match( '/^[a-z0-9-]+\/(v[0-9]+\/)?[a-z0-9-]+$/', $name ) ) {
```

Reply:
You are right, and honestly I want to do that. This allows for things like [version_compare](https://www.php.net/manual/en/function.version-compare.php) being used. If someone doesn't follow that then I guess it could have odd behavior.

This does bring my [deprecation PR](https://github.com/WordPress/wordpress-develop/pull/10507) to mind as well, which _could_ refer to a version.

I feel like there are two directions we could go:
1. No strict boundaries around versioning; just let implementers do their own thing
2. Enforce PHP versioning so we can use versioning functions